zebra-state/src/service/check/difficulty.rs

```diff
@@ -64,14 +64,14 @@ pub(crate) struct AdjustedDifficulty {
     /// The `header.difficulty_threshold`s from the previous
     /// `PoWAveragingWindow + PoWMedianBlockSpan` (28) blocks, in reverse height
     /// order.
-    relevant_difficulty_thresholds: [CompactDifficulty; POW_ADJUSTMENT_BLOCK_SPAN],
+    relevant_difficulty_thresholds: Vec<CompactDifficulty>,
```

Comment:
I was wondering about using `Vec`s, fixed-size arrays of `Option`s, or `AtLeastOne`, any preferences?